File: native-ui/app/src/main/java/com/example/cityinyourpocket/service/UserService.kt
package com.example.cityinyourpocket.service

import android.util.Log
import com.example.cityinyourpocket.model.User

class UserService {
    companion object{
         var users: MutableList<User> = mutableListOf();
    }

    constructor(){
        if(users.size == 0){
            users.add(User("cristi", "parola"))
            users.add(User("mirela", "parola"))
            users.add(User("dana", "parola"))
        }
    }

    fun verifyLogInData(userName: String, password: String): Boolean{
        for(user in users)
            if(user.username == userName && user.password == password)
                return true;
        return false;
    }

    fun userNameIsFree(userName: String): Boolean{
        for(user in users)
            if(user.username == userName)
                return false
        return true
    }

    fun addUser(user: User){
        users.add(user)
        Log.d("UService", "added "+user.username)
    }
}

File: non-native-ui/service/user.service.js
import {UserApp} from './../model/user';

export class UserService{
    static users = [new UserApp("alin","Alin Pintilie", "parola"),
                    new UserApp("mihai","Mihai Ciubotariu", "parola"),
                    new UserApp("dan","Dan Negru", "parola"),
                    new UserApp("dana","Dana Irimies", "parola")]
    constructor(){

    }

    userNameIsTaken(userName){
        let taken = false;
        UserService.users.forEach(value => {
            if(value.username===userName){
                taken = true;
            }
        })
        return taken;
    }

    verifyLogInData(username, password){
        let taken = false;
        UserService.users.forEach(value => {
            if(value.username==username && value.password==password){
                taken = true;
            }
        })
        return taken;
    }
}